zerver/tests/test_narrow.py

```diff
@@ -468,6 +469,68 @@ def test_get_old_messages_with_narrow_sender(self):
         for message in result["messages"]:
             self.assertEqual(message["sender_email"], "othello@zulip.com")
 
+    def test_get_old_messages_with_search(self):
+        self.login("cordelia@zulip.com")
+
+        mutterances = [
+            ('breakfast', 'there are muffins in the conference room'),
+            ('lunch plans', 'I am hungry!'),
+            ('meetings', 'discuss lunch after lunch'),
+            ('meetings', 'please bring your laptops to take notes'),
+            ('dinner', 'Anybody staying late tonight?'),
+        ]
+
+        for topic, content in mutterances:
+            self.send_message(
+                sender_name="cordelia@zulip.com",
+                raw_recipients="Verona",
+                message_type=Recipient.STREAM,
+                content=content,
+                subject=topic,
+            )
+
+        # We use brute force here and update our text search index
+        # for the entire zerver_message table (which is small in test
+        # mode).  In production there is an async process which keeps
+        # the search index up to date.
+        with connection.cursor() as cursor:
+            cursor.execute("""
+                UPDATE zerver_message SET
+                search_tsvector = to_tsvector('zulip.english_us_search',
+                subject || rendered_content)
+                """)
+
+        narrow = [
+            dict(operator='sender', operand='cordelia@zulip.com'),
+            dict(operator='search', operand='lunch'),
+        ]
+        result = self.post_with_params(dict(
+            narrow=ujson.dumps(narrow),
+            use_first_unread_anchor='true',
```

Comment:
why are you using `use_first_unread_anchor` with `num_after` here?  I would have thought the messages would be marked as read (since they were sent by the current user).  I think probably what's going on is this is working by accident due to the `post_with_params` defaults of `num_before=1` and `anchor=1`... probably should clear this up to be clearer...